azext_iot/common/auth.py

```diff
@@ -25,3 +28,21 @@ def get_aad_token(cmd, resource=None):
         "subscription": subscription,
         "tenant": tenant,
     }
+
+
+def generate_device_key(primarykey, device_id):
```

Comment:
In the documentation for [generating a device key](https://docs.microsoft.com/en-us/azure/iot-dps/concepts-symmetric-key-attestation#group-enrollments) the deviceId portion is referred to as a `registrationId` because it can be a composite Id (example uses serial number + MAC address) -- the generic service implementation for this should use the same terminology.